docs/proposals/pod-security-context.md

```diff
@@ -0,0 +1,302 @@
+    individual containers as needed
+
+### Use Case: Pod level security attributes
+
+Some security attributes make sense only to model at the pod level.  For example, it is a
+fundamental property of pods that all containers in a pod share the same network namespace.
+Therefore, using the host namespace makes sense to model at the pod level only, and indeed, today
+it is part of the `PodSpec`.  Other host namespace support is currently being added and these will
+also be pod-level settings; it makes sense to model them as a pod-level collection of security
+attributes.
+
+### Use Case: Default security context for all containers
+
+By default, all containers in a pod should share the same security settings, and those settings
+should only have to be specified once.  If users are forced to specify the same settings for each
+container, it requires more complex validations and allows for transcription errors between
+containers
+
+Additionally, use cases for sharing resources such as volumes are much easier to implement if a user
+can describe a pod-level security context.
+
+### Use Case: Override pod security context for container
+
+Some use cases require the containers in a pod to run with different security settings.  As an
+example, a user may want to have a pod with two containers, one of which runs as root with the
+privileged setting, and one that runs as a non-root UID.  To support use cases like this, it should
+be possible to override appropriate (ie, not intrinsically pod-level) security settings for
+individual containers.
+
+## Analysis
+
+### SELinux context: pod- or container- level?
+
+Currently, SELinux context is specifiable only at the container level.  This is an inconvenient
+factoring for sharing volumes and other SELinux-secured resources between containers because there
+is no way in SELinux to share resources between processes with different MCS labels except to
+remove MCS labels from the shared resource.  This is a big security risk: _any container_ in the
+system can work with a resource which has the same SELinux context as it and no MCS labels.  Since
+we are also not interested in isolating containers in a pod from one another, the SELinux context
+should be shared by all containers in a pod to facilitate isolation from the containers in other
+pods and sharing resources amongst all the containers of a pod.
+
+### Pod-level supplemental groups
+
+The [generalized non-root and SELinux support for volumes](https://github.com/kubernetes/kubernetes/pull/12944)
+pull request describes in detail the cases for sharing volumes using supplemental groups.  The
+easiest way to share resources among containers running as different UID/primary GID combinations is
+to make them owned by a supplemental group that all containers run as.  Therefore, there should be a
+pod-level supplemental group field that cannot be overridden in containers.
+
+### Backward compatibility
```

Comment:
You should link to https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api_changes.md